extension/jdtls.ext/com.microsoft.gradle.bs.importer/src/com/microsoft/java/builder/JavaIncrementalImageBuilder.java

```diff
@@ -0,0 +1,328 @@
+	}
+
+	@Override
+	protected Compiler newCompiler() {
+		// disable entire javadoc support if not interested in diagnostics
+		Map<String, String> projectOptions = ((JavaProblemChecker) this.javaBuilder).javaProject.getOptions(true);
+		String option = projectOptions.get(JavaCore.COMPILER_PB_INVALID_JAVADOC);
+		if (option == null || option.equals(JavaCore.IGNORE)) { // TODO (frederic) see why option is null sometimes while running model tests!?
+			option = projectOptions.get(JavaCore.COMPILER_PB_MISSING_JAVADOC_TAGS);
+			if (option == null || option.equals(JavaCore.IGNORE)) {
+				option = projectOptions.get(JavaCore.COMPILER_PB_MISSING_JAVADOC_COMMENTS);
+				if (option == null || option.equals(JavaCore.IGNORE)) {
+					option = projectOptions.get(JavaCore.COMPILER_PB_UNUSED_IMPORT);
+					if (option == null || option.equals(JavaCore.IGNORE)) { // Unused import need also to look inside javadoc comment
+						projectOptions.put(JavaCore.COMPILER_DOC_COMMENT_SUPPORT, JavaCore.DISABLED);
+					}
+				}
+			}
+		}
+
+		// called once when the builder is initialized... can override if needed
+		CompilerOptions compilerOptions = new CompilerOptions(projectOptions);
+		compilerOptions.performMethodsFullRecovery = true;
+		compilerOptions.performStatementsRecovery = true;
+		Compiler newCompiler = createCompiler(compilerOptions);
+		CompilerOptions options = newCompiler.options;
+		// temporary code to allow the compiler to revert to a single thread
+		String setting = System.getProperty("jdt.compiler.useSingleThread"); //$NON-NLS-1$
+		newCompiler.useSingleThread = setting != null && setting.equals("true"); //$NON-NLS-1$
+
+		// enable the compiler reference info support
+		options.produceReferenceInfo = true;
+
+		if (options.complianceLevel >= ClassFileConstants.JDK1_6
+				&& options.processAnnotations) {
+			// support for Java 6 annotation processors
+			initializeAnnotationProcessorManager(newCompiler);
+		}
+
+		return newCompiler;
+	}
+
+	private Compiler createCompiler(CompilerOptions compilerOptions) {
+		return new Compiler(
+				this.nameEnvironment,
+				DefaultErrorHandlingPolicies.proceedWithAllProblems(),
+				compilerOptions,
+				this,
+				ProblemFactory.getProblemFactory(Locale.getDefault())) {
+					@Override
+					public void accept(ISourceType[] sourceTypes, PackageBinding packageBinding,
+							AccessRestriction accessRestriction) {
+						if (sourceTypes[0] instanceof SourceTypeElementInfo) {
+							// ensure to jump back to toplevel type for first one (could be a member)
+							while (sourceTypes[0].getEnclosingType() != null) {
+								sourceTypes[0] = sourceTypes[0].getEnclosingType();
+							}
+
+							CompilationResult result =
+								new CompilationResult(sourceTypes[0].getFileName(), 1, 1, this.options.maxProblemsPerUnit);
+
+							// https://bugs.eclipse.org/bugs/show_bug.cgi?id=305259, build the compilation unit in its own sand box.
+							final long savedComplianceLevel = this.options.complianceLevel;
```

Comment:
You should always set the `COMPILER_COMPLIANCE` during import, otherwise ECJ will use the embedded JRE version as the default COMPILER_COMPLIANCE. That can cause problems if the actual source level is higher than 17.

According to the definition of COMPILER_COMPLIANCE, COMPILER_SOURCE and COMPILER_CODEGEN_TARGET_PLATFORM settings cannot be higher than the compiler compliance level.

Reply:
@testforstephen here it is https://github.com/microsoft/vscode-gradle/pull/1384